packages/template-manager/internal/template/main.go

```diff
@@ -0,0 +1,52 @@
+package template
+
+import (
+	"context"
+	"fmt"
+
+	artifactregistry "cloud.google.com/go/artifactregistry/apiv1"
+	"cloud.google.com/go/artifactregistry/apiv1/artifactregistrypb"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/e2b-dev/infra/packages/shared/pkg/consts"
+	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
+)
+
+func GetDockerImageURL(templateID string) string {
+	// DockerImagesURL is the URL to the docker images in the artifact registry
+	return fmt.Sprintf("projects/%s/locations/%s/repositories/%s/packages/", consts.GCPProject, consts.GCPRegion, consts.DockerRegistry) + templateID
+}
+
+func Delete(
+	ctx context.Context,
+	tracer trace.Tracer,
+	artifactRegistry *artifactregistry.Client,
+	templateStorage *Storage,
+	buildId string,
+) error {
+	childCtx, childSpan := tracer.Start(ctx, "delete-template")
+	defer childSpan.End()
+
+	err := templateStorage.Remove(ctx, buildId)
+	if err != nil {
+		return fmt.Errorf("error when deleting template objects: %w", err)
+	}
+
+	op, artifactRegistryDeleteErr := artifactRegistry.DeletePackage(ctx, &artifactregistrypb.DeletePackageRequest{Name: GetDockerImageURL(buildId)})
```

Comment:
if it's only snapshot there won't be any image, so notFound is kinda expected here (I would just mention it here as comment or inside the log)